### ISSUE TITLE
clang misleading diagnostic: error: 'f' requires 'float' type support, but ABI 'aapcs' does not support it

### ISSUE THREAD

**Issue by jroelofs** [llvm/llvm-project#90001]
```
float f(void) { return 1.f; }
```
+
```
-mgeneral-regs-only
```
gives:
```
<source>:1:7: error: 'f' requires 'float' type support, but ABI 'aapcs' does not support it
    1 | float f(void) { return 1.f; }
      |       ^
1 error generated.
```

https://clang.godbolt.org/z/d3vGv9ahM

The diagnostic should blame `-mgeneral-regs-only` instead of the ABI, which has been modified by the flag.

**Comment by jroelofs** [comment]
cc @ostannard
